src/runtime_src/core/tools/common/SubCmdConfigureInternal.cpp

```diff
@@ -0,0 +1,462 @@
+  }
+}
+
+static config
+get_daemon_conf()
+{
+  config cfg;
+  cfg.host = xrt_core::utils::get_hostname();
+
+  std::ifstream istr(config_file);
+  if (!istr)
+    return cfg;
+
+  // Load persistent value, may overwrite default one
+  std::string line;
+  while (std::getline(istr, line)) {
+    auto pos = line.find('=', 0);
+    if (pos == std::string::npos)
+      throw xrt_core::system_error(EIO, "Bad daemon config file line '" + line + "'");
+    auto key = line.substr(0, pos++);
+    auto value = line.substr(pos);
+
+    if (key == "host") 
+      cfg.host = value;
+  }
+
+  return cfg;
+}
+
+/*
+ * helper function for option: showx
+ * shows daemon config
+ */
+static void 
+show_daemon_conf()
+{
+  auto cfg = get_daemon_conf();
+  std::cout << "Daemon:\n";
+  std::cout << boost::str(boost::format("  %s\n") % cfg);
+}
+
+/*
+ * helper function for option: showx
+ * shows device config
+ */
+static void 
+show_device_conf(xrt_core::device* device)
+{
+  auto bdf_raw = xrt_core::device_query<xrt_core::query::pcie_bdf>(device);
+  auto bdf = xrt_core::query::pcie_bdf::to_string(bdf_raw);
+
+  std::cout << bdf << "\n";
+
+  bool is_mfg = false;
+  bool is_recovery = false;
+  try {
+    is_mfg = xrt_core::device_query<xrt_core::query::is_mfg>(device);
+    is_recovery = xrt_core::device_query<xrt_core::query::is_recovery>(device);
+  }
+  catch (const std::exception& ex) {
+    std::cerr << ex.what() << "\n";
+  }
+
+  if (is_mfg || is_recovery)
+    throw xrt_core::error(std::errc::operation_canceled, "This operation is not supported with manufacturing image.\n");
+
+  boost::format node_data_format("  %-33s: %s\n");
+  std::string not_supported = "Not supported";
+  std::string sec_level = not_supported;
+  try {
+    sec_level = std::to_string(xrt_core::device_query<xrt_core::query::sec_level>(device));
+  }
+  catch (xrt_core::query::exception&) {
+    //safe to ignore. These sysfs nodes are not present for vck5000 
+  }
+  std::cout << node_data_format % "Security level" % sec_level;
+
+  std::string throttling_enabled = not_supported;
+  try {
+    throttling_enabled = xrt_core::device_query<xrt_core::query::xmc_scaling_enabled>(device);
+  }
+  catch (xrt_core::query::exception&) {
+    //safe to ignore. These sysfs nodes are not present for u30 and vck5000
+  }
+  std::cout << node_data_format % "Clock Throttling enabled" % throttling_enabled;
+
+  std::string throttling_power_override = not_supported;
+  try {
+    throttling_power_override = xrt_core::device_query<xrt_core::query::xmc_scaling_power_override>(device);
+  }
+  catch (xrt_core::query::exception&) {
+    //safe to ignore. These sysfs nodes are not present for u30 and vck5000
+  }
+  std::cout << node_data_format % "Throttling threshold power override" % throttling_power_override;
+
+  std::string throttling_temp_override = not_supported;
+  try {
+    throttling_temp_override = xrt_core::device_query<xrt_core::query::xmc_scaling_temp_override>(device);
```

Comment:
Please replace all of these try catches with the `device_query_default` function call